# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/ConcurrentDelazification.cpp
# Commit: 31d3f2901db9
# Full Hash: 31d3f2901db927a29898c5208a2624b4225382c1
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2024-12-05 09:58:04
# Regressor Bug: 1933681
# File Overlap Count: 1
# Description:
#   Bug 1933681 - Part 1: Allocate InitialStencilAndDelazifications in CompileGlobalScriptToStencilAndMaybeInstantiate. r=nbp
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D230387
# ==============================================================================

diff -r 2694b90c0deb -r 31d3f2901db9 js/src/vm/ConcurrentDelazification.cpp
--- a/js/src/vm/ConcurrentDelazification.cpp	Thu Dec 05 02:39:42 2024 +0200
+++ b/js/src/vm/ConcurrentDelazification.cpp	Wed Dec 04 23:44:33 2024 +0000
@@ -142,10 +142,15 @@
   return true;
 }
 
-bool DelazificationContext::init(const JS::ReadOnlyCompileOptions& options,
-                                 const frontend::CompilationStencil& stencil) {
+bool DelazificationContext::init(
+    const JS::ReadOnlyCompileOptions& options,
+    frontend::InitialStencilAndDelazifications* stencils) {
   using namespace js::frontend;
 
+  // TODO: Rewrite ConcurrentDelazification with
+  //       InitialStencilAndDelazifications.
+  const CompilationStencil& stencil = *stencils->getInitial();
+
   RefPtr<ScriptSource> source(stencil.source);
   DelazificationCache& cache = DelazificationCache::getSingleton();
   if (!cache.startCaching(std::move(source))) {